vm/src/pyobject.rs

```diff
@@ -1242,23 +1238,25 @@ pub trait StaticType {
         crate::builtins::object::PyBaseObject::static_type()
     }
     fn static_type() -> &'static PyTypeRef {
-        static_cell::get(Self::static_cell()).unwrap_or_else(|| unsafe {
-            // SAFETY: object must be initialized by init_* method.
-            // So this is actually not safe as itself.
-            // But easy to find out when it happened in debug build
-            std::hint::unreachable_unchecked()
-        })
+        Self::static_cell()
+            .get()
+            .expect("static type has not been initialized")
```

Comment:
how about keeping `std::hint::unreachable_unchecked` here? this will be frequently called for `PyValue::class` and `expect` will remain None check even it will not fail (unless it has a bug which is easy to be detected in debug mode)

Reply:
I don't think it will be called all that often, since built-in types will implement `PyValue::class` by borrowing from `vm.ctx`, and this is unsound -- somebody could just do `#[pyclass] struct Foo; Foo::static_type()` and get UB. We could make it an unsafe function maybe? Idk, there's still not really a guarantee that an `init` function would be called before calling into Python code that calls into static_type. This is tricky though, cause we're getting into the soundness vs performance aspect of making Rust libraries 😟

Reply:
Ah, that's right. I was forgetting builtin type implementation for that.